pycodestyle.py

```diff
@@ -2125,21 +2125,30 @@ def generate_tokens(self):
             self.report_error(1, 0, 'E902 %s' % self._io_error, readlines)
         tokengen = tokenize.generate_tokens(self.readline)
         try:
+            prev_physical = ''
```

Comment:
Consider 2fc1d07, which re-uses a common 'windowing' pattern to limit the touch points of this change.

Reply:
considered but it's wrong, for instance for blank lines it will run the previous physical line

Reply:
Oh, yuck. Do you mean the `if token[2][0] > self.total_lines`? Since the tests pass with that change, does that suggest the tests are missing an important case?

Reply:
nah, for like:

```console
$ python3 -m tokenize /dev/stdin <<< $'x = 1\n\nx = 2\n'
0,0-0,0:            ENCODING       'utf-8'        
1,0-1,1:            NAME           'x'            
1,2-1,3:            OP             '='            
1,4-1,5:            NUMBER         '1'            
1,5-1,6:            NEWLINE        '\n'           
2,0-2,1:            NL             '\n'           
3,0-3,1:            NAME           'x'            
3,2-3,3:            OP             '='            
3,4-3,5:            NUMBER         '2'            
3,5-3,6:            NEWLINE        '\n'           
4,0-4,1:            NL             '\n'           
5,0-5,0:            ENDMARKER      ''   
```

the `NL` token would physical-check the line before it

Reply:
(either way, I wouldn't consider some (imo very unreadable) itertools over a simple assignment)

Reply:
I'm afraid I don't understand where it goes wrong. Would you be willing to help me understand? From my understanding of the code, the only place this suggested change could go wrong was if `last[4]` raised a `TypeError` due to `last = None`. Is that where it goes wrong? It seems to me in the other cases, the behavior should be the same as with the `prev_physical` variable.

Is there a reason the test suite doesn't include a test that covers this expectation?

Reply:
Fair enough. Nevermind. I'm sure you know what you're doing.